### PR TITLE
[MRG] MAINT circle ci is now more strict with documentation generation

### LINKED ISSUE #862: Avoid documentation rot in continuous integration
I have just added a 'make check' target to the doc Makefile. It fails when there is a warning in the docs.

I have also fixed all the warnings (hours of work :( ).

I think that we should add "make check" as a test on circleci. It needs to be done after we build the docs, as the examples need to be run for warnings to pass.

@jnothman : we might be interested in porting "make check" to scikit-learn, and the CI test too, if the experiment in nilearn is successful. It's been such a fight with warnings in the scikit-learn docs.

### PR DESCRIPTION
**Pull request by aabadie**
Address and fix #862.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
     - "~/nilearn_data"
+
+  pre:
+    # We need to remove conflicting texlive packages.
+    - sudo -E apt-get -yq remove texlive-binaries --purge
+    # Installing required packages for `make -C doc check command` to work.
+    - sudo -E apt-get -yq update
+    - sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install dvipng texlive-latex-base texlive-latex-extra
+    # Force another download for this data
+    - rm -rf /home/ubuntu/nilearn_data/adhd
```
Comment (aabadie) [inline circle.yml]: This line is required in order to have make html-strict work. See issue #869.

Comment (lesteve) [inline circle.yml]: Hmm look like a kludge to me, how do we know the adhd dataset is the one always going to be affected and not any other datasets?

Comment (aabadie) [inline circle.yml]: It's the one used in the failing example from #869 (100% reproducible on CircleCI, if we remove this line).

Comment (GaelVaroquaux) [inline circle.yml]: > > +    - rm -rf /home/ubuntu/nilearn_data/adhd

> Hmm look like a kludge to me, how do we know the adhd dataset is the
> one always going to be affected and not any other datasets?

Yes, it's a kludge. We need to fix the issue. But I'd like to decouple
the two aspects.

Comment (lesteve) [inline circle.yml]: > Yes, it's a kludge. We need to fix the issue

This looks acceptable as a temporary solution, adding a more self-explanatory comment would be good though.

**Comment by aabadie** [comment]
Just a note about this PR: using less subjects when calling `fetch_adhd()` from the Canica resting state example is passing. I tried with 20 but the result image is not satisfactory, I pushed a change where I set this parameter to 30.

File: doc/Makefile [hunk 2]
```diff
@@ -52,6 +52,15 @@ html:
 	@echo
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
 
+html-strict:
+	# These two lines make the build a bit more lengthy, and the
+	# the embedding of images more robust, it's also more strict than the html
```
Comment (lesteve) [inline doc/Makefile]:
```
# These two lines make the build a bit more lengthy, and the
# the embedding of images more robust,
```

This seems like a copy and paste gone wrong, please remove.

**Comment by GaelVaroquaux** [comment]
The documentation now runs well. The examples look OK.

File: Makefile [hunk 3]
```diff
@@ -61,10 +61,13 @@ doc-plot:
 doc:
 	make -C doc html-noplot
 
+.PHONY : doc-check
+doc-check:
+	make -C doc check
```
Comment (GaelVaroquaux) [inline Makefile]: Now that we are using "html-strict", shouldn't we remove this target from the parent Makefile?

**Comment by GaelVaroquaux** [comment]
Aside my comment on the entry in the Makefile that I feel is not useful, I am +1 for merge. Thanks!

**Comment by aabadie** [comment]
> Aside my comment on the entry in the Makefile that I feel is not useful, I am +1 for merge. Thanks!

Will change this.

File: doc/Makefile (hunk 2 above)
Comment (aabadie) [inline doc/Makefile]: I changed the documentation.

File: Makefile (hunk 3 above)
Comment (aabadie) [inline Makefile]: Removed.

**Comment by aabadie** [comment]
The appveyor failure doesn't seem to be related to this PR, I could see the same issue with another PR. I'm wondering if the problem comes from appveyor itself.

Do you think it can be now be merged ?

**Comment by lesteve** [comment]
> The appveyor failure doesn't seem to be related to this PR, I could see the same issue with another PR. I'm wondering if the problem comes from appveyor itself.

Have you tried restarting the AppVeyor build to see whether it was only a temporary glitch? Have you tried to run the tests locally on a Windows machine?

**Comment by lesteve** [comment]
> Have you tried restarting the AppVeyor build to see whether it was only a temporary glitch? Have you tried to run the tests locally on a Windows machine?

At the end of the day, this PR has nothing to do with Windows or the tests so I don't think we should be too worried about a AppVeyor failure.

**Comment by aabadie** [comment]
> Have you tried restarting the AppVeyor build to see whether it was only a temporary glitch? 

I just did it, it's queued and now I hope it won't be cancelled.

> Have you tried to run the tests locally on a Windows machine?

Yes all green.

**Comment by aabadie** [comment]
It seems all is fine now with this PR. Can it be merged ?

**Comment by GaelVaroquaux** [comment]
> It seems all is fine now with this PR. Can it be merged ?

+1 on my side. Thanks a lot

**Comment by lesteve** [comment]
All right, merging, thanks!

**Comment by lesteve** [comment]
@aabadie this may be a one-off thing but next time use a different branch than master to open a PR.

**Pull request merged**